changelog/fragments/helm-conditional.yaml

```
@@ -0,0 +1,27 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      The status conditional `UpdateError` was renamed to `UpgradeError` to better align with Helm nomenclature.
+      This change just affects Helm based-operators.
+
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: yes
+    
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: `UpdateError` status for Helm based-Operator was ranamed 
+      body: >
+        For Helm based-Operator be more aligned with Helm the `UpdateError` was renamed to `UpgradeError`. 
+        Note that it is NOT a breaking change for the Helm based-Operators itself, however, if you have any 
+        script or code implemented which is using this status to perform checks then, will be required replace 
+        `UpdateError` for `UpgradeError` in this places. 
```

Use the yaml `|` operator instead of `>` for the migration body so we can have separate paragraphs for easier readability. We should probably make this change in the fragment template.

Also some general grammar/syntax suggestions.
```suggestion
      body: |
        For Helm-based operators to be more aligned with Helm, the `UpdateError` condition
        reason was renamed to `UpgradeError` for the `ReleaseFailed` condition.

        Note that this is **NOT** a breaking change for Helm-based operators themselves.
        However, any script or code that is depending on this condition reason must be updated
        to use `UpgradeError` instead of `UpdateError`.
```